fierce.py

```diff
@@ -112,6 +112,21 @@ def reverse_query(resolver, ip):
     return query(resolver, dns.reversename.from_address(ip), record_type='PTR')
 
 
+def recursive_query(resolver, domain, record_type='NS'):
+
+    query_domain = str(domain)
```

Comment:
I think we could make this a bit more Pythonic, and use the builtin features of `dnspython` with something like the following:

```python
tld_length = 2
for _ in range(len(domain) - tld_length):
    resp = query(resolver, domain, record_type)
    if resp:
        break
    domain = domain.parent()
else:
    return None

return resp
```